telegram/_chatfullinfo.py

```diff
@@ -19,58 +19,375 @@
 # along with this program.  If not, see [http://www.gnu.org/licenses/].
 """This module contains an object that represents a Telegram ChatFullInfo."""
 from datetime import datetime
-from typing import TYPE_CHECKING, Optional, Sequence
+from typing import TYPE_CHECKING, Optional, Sequence, Tuple
 
 from telegram._birthdate import Birthdate
-from telegram._chat import Chat
+from telegram._chat import Chat, _ChatBase
 from telegram._chatlocation import ChatLocation
 from telegram._chatpermissions import ChatPermissions
 from telegram._files.chatphoto import ChatPhoto
 from telegram._reaction import ReactionType
+from telegram._utils.argumentparsing import parse_sequence_arg
+from telegram._utils.datetime import extract_tzinfo_from_defaults, from_timestamp
 from telegram._utils.types import JSONDict
 
 if TYPE_CHECKING:
-    from telegram import BusinessIntro, BusinessLocation, BusinessOpeningHours, Message
+    from telegram import Bot, BusinessIntro, BusinessLocation, BusinessOpeningHours, Message
 
 
-class ChatFullInfo(Chat):
+class ChatFullInfo(_ChatBase):
     """
     This object contains full information about a chat.
 
     Objects of this class are comparable in terms of equality. Two objects of this class are
     considered equal, if their :attr:`~telegram.Chat.id` is equal.
 
-    Caution:
-        This class is a subclass of :class:`telegram.Chat` and inherits all attributes and methods
-        for backwards compatibility. In the future, this class will *NOT* inherit from
-        :class:`telegram.Chat`.
-
-    .. seealso::
-        All arguments and attributes can be found in :class:`telegram.Chat`.
-
     .. versionadded:: 21.2
 
     Args:
+        id (:obj:`int`): Unique identifier for this chat.
+        type (:obj:`str`): Type of chat, can be either :attr:`PRIVATE`, :attr:`GROUP`,
+            :attr:`SUPERGROUP` or :attr:`CHANNEL`.
+        accent_color_id (:obj:`int`, optional): Identifier of the
+            :class:`accent color <telegram.constants.AccentColor>` for the chat name and
+            backgrounds of the chat photo, reply header, and link preview. See `accent colors`_
+            for more details.
+
+            .. versionadded:: 20.8
         max_reaction_count (:obj:`int`): The maximum number of reactions that can be set on a
             message in the chat.
 
             .. versionadded:: 21.2
+        title (:obj:`str`, optional): Title, for supergroups, channels and group chats.
+        username (:obj:`str`, optional): Username, for private chats, supergroups and channels if
+            available.
+        first_name (:obj:`str`, optional): First name of the other party in a private chat.
+        last_name (:obj:`str`, optional): Last name of the other party in a private chat.
+        is_forum (:obj:`bool`, optional): :obj:`True`, if the supergroup chat is a forum
+            (has topics_ enabled).
+
+            .. versionadded:: 20.0
```

Comment:
I guess we *could* remove all of these versioning directives since `CFI` is actually a new class 🤔 Would also be fine with keeping for completeness

Reply:
would be nice to have a history

Reply:
all right, let's keep then